File: lib/nusantara_lexer/token.dart
import 'package:nusantara/nusantara_lexer/token_type.dart';

class Token {
	final String source;
	final int line;
	final int charPosition;
	final TokenType type;
	final String content;
	Token(this.source, this.line, this.charPosition, this.type, this.content);
}

File: sdk/nusantara/src/token.cc
#include <token.h>

Token::Token(std::string source, const TokenType &type, const int &charIndex,
             std::string value, const int &line)
    : source(std::move(source)), type(type), value(std::move(value)),
      line(line), charIndex(charIndex)
{}

const std::string &Token::getSource() { return this->source; }

TokenType Token::getType() { return this->type; }

const std::string &Token::getValue() { return this->value; }

int Token::getLine() const { return this->line; }

int Token::getCharIndex() const { return this->charIndex; }
